docs/java-rest/high-level/ml/evaluate-data-frame.asciidoc

```diff
@@ -20,14 +20,52 @@ include-tagged::{doc-tests-file}[{api}-request]
 <1> Constructing a new evaluation request
 <2> Reference to an existing index
 <3> The query with which to select data from indices
-<4> Kind of evaluation to perform
-<5> Name of the field in the index. Its value denotes the actual (i.e. ground truth) label for an example. Must be either true or false
-<6> Name of the field in the index. Its value denotes the probability (as per some ML algorithm) of the example being classified as positive
-<7> The remaining parameters are the metrics to be calculated based on the two fields described above.
-<8> https://en.wikipedia.org/wiki/Precision_and_recall[Precision] calculated at thresholds: 0.4, 0.5 and 0.6
-<9> https://en.wikipedia.org/wiki/Precision_and_recall[Recall] calculated at thresholds: 0.5 and 0.7
-<10> https://en.wikipedia.org/wiki/Confusion_matrix[Confusion matrix] calculated at threshold 0.5
-<11> https://en.wikipedia.org/wiki/Receiver_operating_characteristic#Area_under_the_curve[AuC ROC] calculated and the curve points returned
+<4> Evaluation to be performed
+
+==== Evaluation
+
+Evaluation to be performed.
+Currently, supported evaluations include: +BinarySoftClassification+, +Classification+, +Regression+.
+
+===== Binary soft classification
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-evaluation-softclassification]
+--------------------------------------------------
+<1> Constructing a new evaluation
+<2> Name of the field in the index. Its value denotes the actual (i.e. ground truth) label for an example. Must be either true or false.
+<3> Name of the field in the index. Its value denotes the probability (as per some ML algorithm) of the example being classified as positive.
+<4> The remaining parameters are the metrics to be calculated based on the two fields described above.
+<5> https://en.wikipedia.org/wiki/Precision_and_recall[Precision] calculated at thresholds: 0.4, 0.5 and 0.6
+<6> https://en.wikipedia.org/wiki/Precision_and_recall[Recall] calculated at thresholds: 0.5 and 0.7
+<7> https://en.wikipedia.org/wiki/Confusion_matrix[Confusion matrix] calculated at threshold 0.5
+<8> https://en.wikipedia.org/wiki/Receiver_operating_characteristic#Area_under_the_curve[AuC ROC] calculated and the curve points returned
+
+===== Classification
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-evaluation-classification]
+--------------------------------------------------
+<1> Constructing a new evaluation
+<2> Name of the field in the index. Its value denotes the actual (i.e. ground truth) class the example belongs to.
+<3> Name of the field in the index. Its value denotes the predicted (as per some ML algorithm) class of the example.
+<4> The remaining parameters are the metrics to be calculated based on the two fields described above.
+<5> Multiclass confusion matrix of size 3
```

Comment:
nit: some of these numbered points end with a full stop and others don't.  @szabosteve is there a best practice for this?

(Same in other lists of numbered items.)

Reply:
@przemekwitek @droberts195 I don't think there is a consensus about this in the ES documentation. In most cases, when a list item consists of only one sentence, there is no full stop at the end (which burns my eyes). The only important thing is to remain consistent, so the convention that you described Przemek is great I think. Thank you for paying attention to this.